clang/test/dpct/helper_files_ref/include/rng_utils.hpp

```diff
@@ -174,6 +176,189 @@ template <typename engine_t> class rng_generator {
+  if (dimensions == _dimensions) {
+    return;
+  }
+  destroy();
+  _dimensions = dimensions;
+  create();
+}
+
+void create(const sycl::queue &queue) {
+  if (queue == _queue) {
+    return;
+  }
+  destroy();
+  _queue = queue;
+  create();
+}
+
+void destroy() {
+  _philox4x32x10_engine.reset();
+  _mrg32k3a_engine.reset();
+  _mt2203_engine.reset();
+  _mt19937_engine.reset();
+  _sobol_engine.reset();
+}
+
+inline void generate_uniform_bits(unsigned int *outputPtr, std::int64_t n) {
+   generate(_distr_uniform_bits_uint32_t, (uint32_t *)outputPtr, n);
+}
+
+inline void generate_uniform_bits(unsigned long long *outputPtr, std::int64_t n) {
+   generate(_distr_uniform_bits_uint64_t, (uint64_t *)outputPtr, n);
+}
+
+inline void generate_lognormal(float *outputPtr, std::int64_t n) {
+   generate(_distr_lognormal_float, outputPtr, n);
+}
+
+inline void generate_lognormal(double *outputPtr, std::int64_t n) {
+   generate(_distr_lognormal_double, outputPtr, n);
+}
+
+inline void generate_gaussian(float *outputPtr, std::int64_t n) {
+   generate(_distr_gaussian_float, outputPtr, n);
+}
+
+inline void generate_gaussian(double *outputPtr, std::int64_t n) {
+   generate(_distr_gaussian_double, outputPtr, n);
+}
+
+inline void generate_poisson(unsigned int *outputPtr, std::int64_t n) {
+   generate(_distr_poisson_int32_t, (int32_t *)outputPtr, n);
+}
+
+inline void generate_uniform(float *outputPtr, std::int64_t n) {
+   generate(_distr_uniform_float, outputPtr, n);
+}
+
+inline void generate_uniform(double *outputPtr, std::int64_t n) {
+   generate(_distr_uniform_double, outputPtr, n);
+}
+
+void skip_ahead(const std::uint64_t &num_to_skip) {
+  switch (_type) {
+  case random_engine_type::philox4x32x10:
+    oneapi::mkl::rng::skip_ahead(*_philox4x32x10_engine, num_to_skip);
+    break;
+  case random_engine_type::mrg32k3a:
+    oneapi::mkl::rng::skip_ahead(*_mrg32k3a_engine, num_to_skip);
+    break;
+  case random_engine_type::mt2203:
+    // TODO: no skip_ahead.
+    break;
+  case random_engine_type::mt19937:
+    oneapi::mkl::rng::skip_ahead(*_mt19937_engine, num_to_skip);
+    break;
+  case random_engine_type::sobol:
+    oneapi::mkl::rng::skip_ahead(*_sobol_engine, num_to_skip);
+    break;
+  }
+}
+
+private:
+void create() {
+  switch (_type) {
+  case random_engine_type::philox4x32x10:
+    _philox4x32x10_engine =
+        std::make_shared<oneapi::mkl::rng::philox4x32x10>(_queue, _seed);
+    break;
+  case random_engine_type::mrg32k3a:
+    _mrg32k3a_engine =
+        std::make_shared<oneapi::mkl::rng::mrg32k3a>(_queue, _seed);
+    break;
+  case random_engine_type::mt2203:
+    _mt2203_engine = std::make_shared<oneapi::mkl::rng::mt2203>(_queue, _seed);
+    break;
+  case random_engine_type::mt19937:
+    _mt19937_engine =
+        std::make_shared<oneapi::mkl::rng::mt19937>(_queue, _seed);
+    break;
+  case random_engine_type::sobol:
+    _sobol_engine =
+        std::make_shared<oneapi::mkl::rng::sobol>(_queue, _dimensions);
```

Comment:
This call is using dimensions, other calls are using seed. Seems like an error

Reply:
The same reply as above one.